File: Principal/Html/Componente/ArquivoViewer.cs
using Drive.Html.Componente.Detalhe;
using NetZ.Web.Html;
using NetZ.Web.Html.Componente;
using NetZ.Web.Server.Arquivo.Css;

namespace Drive.Html.Componente
{
    internal class ArquivoViewer : ComponenteHtml
    {
        #region Constantes

        #endregion Constantes

        #region Atributos

        private Div _divConteudo;
        private Div _divVazio;

        private Div divConteudo
        {
            get
            {
                if (_divConteudo != null)
                {
                    return _divConteudo;
                }

                _divConteudo = new Div();

                return _divConteudo;
            }
        }

        private Div divVazio
        {
            get
            {
                if (_divVazio != null)
                {
                    return _divVazio;
                }

                _divVazio = new Div();

                return _divVazio;
            }
        }

        #endregion Atributos

        #region Construtores

        #endregion Construtores

        #region Métodos

        protected override void addLayoutFixo(JavaScriptTag tagJs)
        {
            base.addLayoutFixo(tagJs);

            tagJs.addLayoutFixo(typeof(ArqDetalheGeral));
            tagJs.addLayoutFixo(typeof(ArquivoTile));
        }

        protected override bool getBooJs()
        {
            return true;
        }

        protected override void inicializar()
        {
            base.inicializar();

            this.strId = this.GetType().Name;

            this.divVazio.strConteudo = "Nada por aqui Jhoe.";
        }

        protected override void montarLayout()
        {
            base.montarLayout();

            this.divVazio.setPai(this);
            this.divConteudo.setPai(this);
        }

        protected override void setCss(CssArquivo css)
        {
            base.setCss(css);

            this.addCss(css.setPaddingTop(90));

            this.divVazio.addCss(css.setBackgroundImage("http://s11.postimg.org/j43gatslb/ghost.png"));
            this.divVazio.addCss(css.setBackgroundPosition("center 0px"));
            this.divVazio.addCss(css.setBackgroundRepeat("no-repeat"));
            this.divVazio.addCss(css.setBackgroundSize("100px"));
            this.divVazio.addCss(css.setDisplay("none"));
            this.divVazio.addCss(css.setHeight(200));
            this.divVazio.addCss(css.setLineHeight(250));
            this.divVazio.addCss(css.setMarginTop(100));
            this.divVazio.addCss(css.setTextAlign("center"));
        }

        protected override void setStrId(string strId)
        {
            base.setStrId(strId);

            this.divConteudo.strId = (strId + "_divConteudo");
            this.divVazio.strId = (strId + "_divVazio");
        }

        #endregion Métodos

        #region Eventos

        #endregion Eventos
    }
}

File: Drive.TypeScript/html/componente/ArquivoViewer.ts
/// <reference path="../../../Web.TypeScript/html/componente/ComponenteHtml.ts"/>
/// <reference path="../../../Web.TypeScript/html/Div.ts"/>

module Drive
{
    // #region Importações

    import ComponenteHtml = Web.ComponenteHtml;
    import Div = Web.Div;

    // #endregion Importações

    // #region Enumerados
    // #endregion Enumerados

    export class ArquivoViewer extends ComponenteHtml
    {
        // #region Constantes
        // #endregion Constantes

        // #region Atributos

        private _arrArq: Array<ArquivoDominio>;
        private _divConteudo: Div;
        private _divVazio: Div;

        private get arrArq(): Array<ArquivoDominio>
        {
            return this._arrArq;
        }

        private set arrArq(arrArq: Array<ArquivoDominio>)
        {
            this._arrArq = arrArq;
        }

        private get divConteudo(): Div
        {
            if (this._divConteudo != null)
            {
                return this._divConteudo;
            }

            this._divConteudo = new Div(this.strId + "_divConteudo");

            return this._divConteudo;
        }

        private get divVazio(): Div
        {
            if (this._divVazio != null)
            {
                return this._divVazio;
            }

            this._divVazio = new Div(this.strId + "_divVazio");

            return this._divVazio;
        }

        // #endregion Atributos

        // #region Construtores

        constructor()
        {
            super(ArquivoViewer.name);
        }

        // #endregion Construtores

        // #region Métodos

        private addArquivo(arq: ArquivoDominio): void
        {
            if (arq == null)
            {
                return;
            }

            var divArq = new ArquivoTile(this.arrArq.indexOf(arq), arq);

            this.divConteudo.jq.append(divArq.strLayoutFixo);

            divArq.iniciar();
        }

        private atualizarConteudo(): void
        {
            this.divConteudo.strConteudo = null;

            if (this.arrArq == null)
            {
                this.divVazio.mostrar();
                return;
            }

            if (this.arrArq.length < 1)
            {
                this.divVazio.mostrar();
                return;
            }

            this.divVazio.esconder();

            this.arrArq.forEach((arq) => { this.addArquivo(arq) });
        }

        public carregarConteudo(arrArq: Array<ArquivoDominio>): void
        {
            if (arrArq == null)
            {
                return;
            }

            if (arrArq.length < 1)
            {
                return;
            }

            this.arrArq = arrArq;

            this.atualizarConteudo();
        }

        public carregarConteudoVazio(): void
        {
            this.arrArq = null;

            this.atualizarConteudo();
        }

        public limparConteudo(): void
        {
            this.arrArq == null;
            this.divConteudo.strConteudo = null;
        }

        // #endregion Métodos

        // #region Eventos
        // #endregion Eventos
    }
}